examples/pytorch/graphsaint/README.md

```diff
@@ -26,74 +26,120 @@ All datasets used are provided by Author's [code](https://github.com/GraphSAINT/
+
+> Note that the sampling method in pre-sampling phase must be offline sampling.
 
 ### F1-micro
 
 #### Random node sampler
 
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Paper | 0.960±0.001 | 0.507±0.001 |
-| Running | 0.9628 | 0.5077 |
-| DGL | 0.9618 | 0.4828 |
+| Method | PPI | Flickr | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Paper | 0.960±0.001 | 0.507±0.001 | 0.962±0.001 | 0.641±0.000 | 0.782±0.004 |
+| Running | 0.9628 | 0.5077 | 0.9622 | 0.6393 | 0.7695 |
+| DGL_offline | 0.9715      | 0.4445      | 0.9645 | 0.6457 | 0.8051 |
+| DGL_online | 0.9730 | 0.4358 | 0.9645 | 0.6444 | 0.8014 |
 
 #### Random edge sampler
 
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Paper | 0.981±0.007 | 0.510±0.002 |
-| Running | 0.9810 | 0.5066 |
-| DGL | 0.9818 | 0.5054 |
+| Method      | PPI         | Flickr      | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Paper | 0.981±0.007 | 0.510±0.002 | 0.966±0.001 | 0.653±0.003 | 0.807±0.001 |
+| Running | 0.9810 | 0.5066 | 0.9656 | 0.6531 | 0.8071 |
+| DGL_offline | 0.9817      | 0.4456      | 0.9655 | 0.6530 | 0.8034 |
+| DGL_online | 0.9815 | 0.4508 | 0.9653 | 0.6516 | 0.7756 |
 
 #### Random walk sampler
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Paper | 0.981±0.004 | 0.511±0.001 |
-| Running | 0.9812 | 0.5104 |
-| DGL | 0.9818 | 0.5018 |
+| Method      | PPI         | Flickr      | Reddit      | Yelp        | Amazon      |
+| --- | --- | --- | --- | --- | --- |
+| Paper | 0.981±0.004 | 0.511±0.001 | 0.966±0.001 | 0.653±0.003 | 0.815±0.001 |
+| Running | 0.9812 | 0.5104 | 0.9648      | 0.6527      | 0.8131      |
+| DGL_offline | 0.9833      | 0.4592      | 0.9582      | 0.6514      | 0.8178   |
+| DGL_online | 0.9820 | 0.4606 | 0.9572      | 0.6508      | 0.8157   |
 
 ### Sampling time
 
+- Here sampling time includes consumed time of pre-sampling subgraphs and calculating normalization coefficients in the beginning.
+
 #### Random node sampler
 
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Sampling(Running) | 0.77 | 0.65 |
-| Sampling(DGL) | 0.24 | 0.57 |
-| Normalization(Running) | 0.69 | 2.84 |
-| Normalization(DGL) | 1.04 | 0.41 |
+| Method      | PPI  | Flickr | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Running | 1.46 | 3.49 | 19 | 59.01 | 978.62 |
+| DGL_offline | 2.51 | 1.12 | 27.32 | 60.15 | 929.24 |
+| DGL_online | 3.17 | 1.11 | 27.82 | 59.42 | 991.68 |
 
 #### Random edge sampler
 
-| Method | PPI | Flickr |
-| --- | --- | --- |
-| Sampling(Running) | 0.72 | 0.56 |
-| Sampling(DGL) | 0.50 | 0.72 |
-| Normalization(Running) | 0.68 | 2.62 |
-| Normalization(DGL) | 0.61 | 0.38 |
+| Method      | PPI  | Flickr | Reddit | Yelp | Amazon |
+| --- | --- | --- | --- | --- | --- |
+| Running | 1.4 | 3.18 | 13.88 | 39.02 | 164.35 |
+| DGL_offline | 3.08 | 1.91 | 53.59 | 48.94 | 1469.52 |
+| DGL_online | 3.04 | 1.87 | 52.01 | 48.38 | 1432.79 |
```

Comment:
Have you tried the proposal on sampling without replacement?

Reply:
Nope lol if it's necessary I can take a try later.

Reply:
Yes, please have a try and see if we can simply perform sampling without replacement in that case.